src/porepy/numerics/mixed_dim/assembler.py

```diff
@@ -1202,3 +1204,120 @@ def num_dof(self):
             int: Number of unknowns. Size of solution vector.
         """
         return self.full_dof.sum()
+
+    def variables_of_grid(
+        self, g: Union[pp.Grid, Tuple[pp.Grid, pp.Grid]]
+    ) -> List[str]:
+        """ Get all variables defined for a given grid or edge.
+
+        Args:
+            g (Union[pp.Grid, Tuple[pp.Grid, pp.Grid]]): Target grid, or an edge
+
+        Returns:
+            List[str]: List of all variables known for this entity.
+
+        """
+        names = []
+        for key in self.block_dof.keys():
+            if key[0] == g:
+                names.append(key[1])
+        return names
+
+    def __str__(self) -> str:
+        s = f"Assembler object on a GridBucket with {self.gb.num_graph_nodes()} "
+        s += f"subdomains and {self.gb.num_graph_edges()} interfaces.\n"
+        s += f"Total number of degrees of freedom: {self.num_dof()}\n"
+        s += f"Total number of subdomain and interface variables: {len(self.block_dof)}\n"
+
+        names = []
+        for key in self.block_dof.keys():
+            names.append(key[1])
+
+        unique_vars = list(set(names))
+        s += f"Variable names {unique_vars}"
+
+        return s
+
+    def __repr__(self) -> str:
+        s = f"Assembler object with in total {self.num_dof()} dofs"
+        s += f" on {len(self.block_dof)} subdomain and interface variables.\n"
+
+        s += f"Maximum grid dimension: {self.gb.dim_max()}.\n"
+        s += f"Minimum grid dimension: {self.gb.dim_min()}.\n"
+
+        for dim in range(self.gb.dim_max(), self.gb.dim_min() - 1, -1):
+            s += f"In dimension {dim}: {len(self.gb.grids_of_dimension(dim))} grids.\n"
+            names = []
+            for key in self.block_dof.keys():
+                if isinstance(key[0], tuple):
+                    continue
+                elif key[0].dim == dim:
+                    names.append(key[1])
+
+            unique_vars = list(set(names))
+            s += f"All variables present in dimension {dim}: {unique_vars}\n"
+
+            # Also check if some subdomains of this dimension have a subset of the
+            # variables defined on the totality of the subdomains
```

Comment:
I see the danger. My idea was to print each variable combination once, so that if you have thm, but hm in a subset of fractures, hm will be printed once. We can also drop this altogether, but I can see this becoming really useful for debugging. 

Opinions?

Reply:
Yes, I agree that your approach could be useful. My suggestion is something like:
combination_0: node_number_set_0
combination_1: node_number_set_1
...
E.g.
TH + contact: 1, 2   # (some fractures)
H + contact:   3, 4   # other fractures where you don't have T

In this way, you know exactly where each combination applies

Reply:
That would indeed be useful. I'm not too fund of node_numbers, since these are somewhat randomly set, at least for fracture intersections. On the other hand, it would at least give a point to start looking for trouble - if this information is needed, it is most likely in a debugging scenario.